mailbox/store/src/main/java/org/apache/james/mailbox/store/event/EventFactory.java

```diff
@@ -192,38 +194,39 @@ public Mailbox getMailbox() {
         }
     }
 
-    public MailboxListener.Added added(MailboxSession session, SortedMap<MessageUid, MessageMetaData> uids, Mailbox mailbox, Map<MessageUid, MailboxMessage> cachedMessages) {
-        return new AddedImpl(session, mailbox, uids, cachedMessages);
+    public MailboxListener.Added added(Optional<MailboxSession.SessionId> sessionId, User user, SortedMap<MessageUid, MessageMetaData> uids, Mailbox mailbox, Map<MessageUid, MailboxMessage> cachedMessages) {
+        return new AddedImpl(sessionId, user, mailbox, uids, cachedMessages);
     }
 
-    public MailboxListener.Expunged expunged(MailboxSession session,  Map<MessageUid, MessageMetaData> uids, Mailbox mailbox) {
-        return new ExpungedImpl(session, mailbox, uids);
+    public MailboxListener.Expunged expunged(Optional<MailboxSession.SessionId> sessionId, User user,  Map<MessageUid, MessageMetaData> uids, Mailbox mailbox) {
+        return new ExpungedImpl(sessionId, user, mailbox, uids);
     }
 
-    public MailboxListener.FlagsUpdated flagsUpdated(MailboxSession session, List<MessageUid> uids, Mailbox mailbox, List<UpdatedFlags> uflags) {
-        return new FlagsUpdatedImpl(session, mailbox, uids, uflags);
+    public MailboxListener.FlagsUpdated flagsUpdated(Optional<MailboxSession.SessionId> sessionId, User user, List<MessageUid> uids, Mailbox mailbox, List<UpdatedFlags> uflags) {
+        return new FlagsUpdatedImpl(sessionId, user, mailbox, uids, uflags);
     }
 
-    public MailboxListener.MailboxRenamed mailboxRenamed(MailboxSession session, MailboxPath from, Mailbox to) {
-        return new MailboxRenamedEventImpl(session, from, to);
+    public MailboxListener.MailboxRenamed mailboxRenamed(Optional<MailboxSession.SessionId> sessionId, User user, MailboxPath from, Mailbox to) {
+        return new MailboxRenamedEventImpl(sessionId, user, from, to);
     }
 
-    public MailboxListener.MailboxDeletion mailboxDeleted(MailboxSession session, Mailbox mailbox, QuotaRoot quotaRoot,
+    public MailboxListener.MailboxDeletion mailboxDeleted(Optional<MailboxSession.SessionId> sessionId, User user, Mailbox mailbox, QuotaRoot quotaRoot,
```

Comment:
I don't agree you removed the factory methods. It's very useful to create an event from a MailboxSession. It makes the event creation more painful than needed.

Reply:
There are some places, passing Optional as method param is convenience, I keep them, and duplicate to a new method which take non Optional param. see https://github.com/linagora/james-project/pull/1980/commits/5936f9811afcda6d93dbe0bba484ff9bf1df4464

Reply:
you misread my comment, I was talking about keeping previous signature as a convenient way to create the event (so accept MailboxSession)

Reply:
revert old factory methods in 7bf952e